### DISCUSSION TITLE
How would I pass props into pinia store?

### DISCUSSION THREAD

**Discussion by shadowmkj** [vuejs/pinia#1266] [Help and Questions]
I am using vue3 composition API with InertiaJS integration. I have some objects that I receive from controllers using
`props: {
       products: Object
   }`

I want to pass the products prop into my Pinia Store. Is there any way? 
Using composables I could do `const store = useStore(props)`
But this is not possible using Pinia store. 
Please respond if you know a way to tackle this.

**Comment by MickL** [comment]
2 years later, no one has an answer? :(

I want to create a store where I can decide weither the data is fetched on server side or on client side:

```ts
export const useMyStore = defineStore('my-store', (server: boolean) => {
  const { pending, data, error } = useLazyFetch<MyResponse>('/api/my-api', {server});

  return {
    // State
    data,
    pending,
    error,

    // Getters

    // Actions
  };
});
```

**Reply by ilyarefraction** [reply]
Create a store factory function like below

```js
import { defineStore } from 'pinia';

export default (options = { useServer: true }) => {
  return defineStore('my-store', () => {
    const { pending, data, error } = useLazyFetch('/api/my-api', { server: options.useServer });

    return {
      data,
      pending,
      error,
    };
  })();
};
```
If you need different stores with the same structure (i.e. one to work with client and one to work with server) you can add namespaces like this.

```js
export default (namespace = 'serverStore', options = { useServer: true }) => {
  return defineStore(namespace, () => {
    const { pending, data, error } = useLazyFetch('/api/my-api', { server: options.useServer });

    return {
      data,
      pending,
      error,
    };
  })();
};
```

**Comment by mazenbassiouni** [comment]
This was helpful in my case, while refreshing the state `myStore.refreshProducts()` every time products are possible to get changed
```
import { defineStore } from 'pinia'
import { usePage } from '@inertiajs/vue3'

export const useMyStore  = defineStore({
    state: () => ({
        products: usePage().props.products
    }),
    actions: {
        refreshProducts(){
            this.products = usePage().props.products
        }
    }
})  

```
